c/ovm.c

```diff
@@ -515,23 +522,18 @@ static word onum(int64_t a) {
       x = -a;
    }
    if (x > FMAX) {
-      word *p, l = INULL, v = x >> (2 * FBITS);
-      if (v != 0) {
+      word *p = (word *)INULL;
+      unsigned int shift = (63 / FBITS) * FBITS;
+      while (!(x & ((uint64_t)FMAX << shift)))
```

Comment:
This is a nice improvement over the old one, and scales properly if fixnum sizes are modified later.